src/modules/claims/claims.service.ts

```diff
@@ -1354,22 +1358,40 @@ export class ClaimsService {
     );
   }
 
-  /**
+    /**
+   * Verify the issuer and chain of trust for a Verifiable Credential (VC)
+   *
+   * @param {VerifiableCredential<RoleCredentialSubject} vc to be verified
+   */
+     async verifyVc(vc: VerifiableCredential<RoleCredentialSubject>) {
+      const issuerDID = this._signerService.did;
+      const role = vc.credentialSubject.role.namespace;
+      if (
+        !(
+          (await this._vcIssuerVerifier.verifyIssuerAuthority(role, issuerDID)) ||
+          (await this._vcIssuerVerifier.verifyChainOfTrustByRoleDefinition(vc))
+        )
+      ) {
+        throw new NotAuthorizedIssuer(issuerDID, role);
```

Comment:
`verifyIssuance` and `verifyIssuer` throws errors with descriptive message. Debugging would be easier if we attached this messages to message of NotAuthorizedIssuer. Perhaps we don't even need this NotAuthorizedIssuer

Reply:
@JGiter How do you want to handle this

Reply:
I would just verify these two one by one
```
const role = vc.credentialSubject.role.namespace;
await this._vcIssuerVerifier.verifyIssuance(issuerDID, role);
await this._vcIssuerVerifier.verifyIssuer(issuerDID, role);
```

Reply:
No, I think we should not do `await this._vcIssuerVerifier.verifyIssuance(issuerDID, role);`, as this requires that the credential be obtained external (e.g. from IPFS) https://github.com/energywebfoundation/ew-credentials/blob/c54672118a2687146e8421d3779208a04ee191c0/packages/vc-verification/src/issuer-verification-vc.ts#L68

We should instead verify the proof using VC service as this can verify the credential directly https://energy-web-foundation-iam-client-lib.readthedocs-hosted.com/api/classes/modules_verifiable_credentials.VerifiableCredentialsServiceBase/#verify

So it should be more like
```
if (! await verifiableCredentialsService.verify(credential)) {
 throw ProofNotVerifiedError()
}
const role = vc.credentialSubject.role.namespace;
await this._vcIssuerVerifier.verifyIssuer(issuerDID, role);
```
(the above almost certainly needs to be tweaked but I tink it is the better direction)